app/models/namespace.rb

```diff
@@ -175,6 +175,10 @@ def children_allowed?
     ancestors.count >= Namespace::MAX_ANCESTORS - 2
   end
 
+  def direct_descendants_of_type?(type)
```

Comment:
**nitpick (blocking):** Since I am asking you to create a method called `children_of_type` can you rename this as `children_of_type?`. The new method will execute the `where` where as this method just uses `exists` and has a the `?` to indicate boolean response. Maybe we rename this to `has_children_of_type?` instead to make it more obvious.

Reply:
Updated in c7c240a